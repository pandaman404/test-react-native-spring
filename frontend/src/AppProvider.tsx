import { NavigationContainer } from '@react-navigation/native';
import { StatusBar } from 'expo-status-bar';
import { ReactNode } from 'react';
import { SafeAreaProvider } from 'react-native-safe-area-context';

interface AppProviderProps {
  children: ReactNode;
}

export default function AppProvider({ children }: AppProviderProps) {
  return (
    <SafeAreaProvider>
      <NavigationContainer>
        <StatusBar style='auto' />
        {children}
      </NavigationContainer>
    </SafeAreaProvider>
  );
}
