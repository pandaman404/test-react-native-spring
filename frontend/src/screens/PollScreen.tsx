import { View, StyleSheet } from 'react-native';
import React from 'react';
import { SafeAreaView } from 'react-native-safe-area-context';
import GoBackBtn from '../components/GoBackBtn';
import Title from '../components/Title';
import { Controller } from 'react-hook-form';
import SelectField from '../components/SelectField';
import TextField from '../components/TextField';
import PressBtn from '../components/PressBtn';
import ErrorMessage from '../components/ErrorMessage';
import { mapEnumToOptions } from '../utility/mapEnumToOptions';
import { Colors } from 'react-native/Libraries/NewAppScreen';
import { MusicGenre } from '../@types/MusicGenre.enum';
import { useHandlePollForm } from '../hooks/useHandlePollForm';

export default function PollScreen() {
  const { control, handleSubmit, errors, onSubmit, isSubmitting, remoteErrorMessage } = useHandlePollForm();
  return (
    <SafeAreaView style={styles.container}>
      <GoBackBtn />
      <View style={styles.formContainer}>
        <Title text='Encuesta' />
        <Controller
          control={control}
          render={({ field: { onChange } }) => (
            <SelectField
              onChange={onChange}
              items={mapEnumToOptions(MusicGenre)}
              placeholder='Seleccione estilo mÃºsical'
              error={errors.musicGenre?.message}
            />
          )}
          name='musicGenre'
        />
        <Controller
          control={control}
          render={({ field: { onChange, onBlur, value } }) => (
            <TextField
              value={value}
              onChange={onChange}
              onBlur={onBlur}
              placeholder='johndoe@email.org'
              error={errors.email?.message}
            />
          )}
          name='email'
        />
        <PressBtn title='Enviar' onPress={handleSubmit(onSubmit)} isLoading={isSubmitting} />
        <View>{remoteErrorMessage.length > 0 && <ErrorMessage text={remoteErrorMessage} />}</View>
      </View>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    gap: 40,
    backgroundColor: Colors.white,
  },
  formContainer: {
    justifyContent: 'center',
    alignItems: 'center',
    gap: 30,
    width: '75%',
    alignSelf: 'center',
    position: 'relative',
  },
});
