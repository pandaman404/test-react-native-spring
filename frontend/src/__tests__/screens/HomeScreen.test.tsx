import renderer from 'react-test-renderer';
import AppProvider from '../../AppProvider';
import PollScreen from '../../screens/PollScreen';

describe('<PollScreen />', () => {
  it('Should render PollScreen', () => {
    const tree = renderer
      .create(
        <AppProvider>
          <PollScreen />
        </AppProvider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
