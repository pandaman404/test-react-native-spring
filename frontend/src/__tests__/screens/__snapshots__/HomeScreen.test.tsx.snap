// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PollScreen /> Should render PollScreen 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
